bundles/remote_services/remote_services_api/include/remote_custom_serialization_service.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CELIX_REMOTE_CUSTOM_SERIALIZATION_SERVICE_H
```

Comment:
This header has a TODO to add serialize/deserialize to it as well, does that make sense?
@pnoltes I think it does, just want to make sure we don't overlook something. Is it, for example, possible to not use a serialization solution/service?
Also, currently serialization is not in a service at all, if we do this, I think it makes sense to move it asap as well. Current situation could just be a wrapper for the used json_rpc I think.
Also, if we do it like this for the RSA, I think a similar solution is needed for the PSA, wdyt?

Reply:
@Oipo & @abroekhuis  WDYT?

Note that this is for now too much work for this PR. Maybe support the above mentioned rsa_custom_serialization_service and create a ticker for the pubsub?

Reply:
How does this relate to the existing serializers already in pubsub? Most is already in place there, right?
Other than that, I think this sounds like a good approach. If we can solve the RSA part in this PR, that would be great!